packages/lodestar/src/sync/backfill/backfill.ts

```diff
@@ -0,0 +1,336 @@
+    super();
+    this.chain = modules.chain;
+    this.network = modules.network;
+    this.db = modules.db;
+    this.config = modules.config;
+    this.logger = modules.logger;
+    this.metrics = modules.metrics;
+    this.wsCheckpoint = modules.wsCheckpoint;
+    this.opts = opts ?? {batchSize: BATCH_SIZE};
+    this.network.events.on(NetworkEvent.peerConnected, this.addPeer);
+    this.network.events.on(NetworkEvent.peerDisconnected, this.removePeer);
+
+    this.sync()
+      .then((oldestSlotSynced) => {
+        this.emit(BackfillSyncEvent.completed, oldestSlotSynced);
+        this.logger.info("BackfillSync completed", {oldestSlotSynced});
+        this.status = BackfillSyncStatus.completed;
+        // Sync completed, unsubscribe listeners and don't run the processor again
+        // Backfill is never necessary again until the node shuts down
+        this.close();
+      })
+      .catch((e) => {
+        if (e instanceof ErrorAborted) {
+          return; // Ignore
+        }
+        this.logger.error("BackfillSync processor error", e);
+        this.status = BackfillSyncStatus.completed;
+        this.close();
+      });
+    if (this.metrics) {
+      this.metrics.backfillSync.status.addCollect(() => this.metrics?.backfillSync.status.set(syncStatus[this.status]));
+    }
+  }
+
+  /** Throw / return all AsyncGenerators */
+  close(): void {
+    this.network.events.off(NetworkEvent.peerConnected, this.addPeer);
+    this.network.events.off(NetworkEvent.peerDisconnected, this.removePeer);
+    this.processor.end(new ErrorAborted("BackfillSync"));
+  }
+
+  /** Returns oldestSlotSynced */
+  private async sync(): Promise<Slot> {
+    // Start processor loop
+    this.processor.trigger();
+    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+    const checkpointSlot = (this.wsCheckpoint?.epoch || 0) * SLOTS_PER_EPOCH;
+
+    for await (const _ of this.processor) {
+      if (!this.anchorBlock && !this.anchorBlockRoot) {
+        // In case of first run or an irrecoverable error, both anchorBlock and anchorBlockRoot could be nullified in order to start Backfill sync fresh!
+        const anchorCp = this.chain.forkChoice.getFinalizedCheckpoint();
+        // Look in db, as finalized root might not have moved at all on a quick restart
+        this.anchorBlock = await this.db.blockArchive.getByRoot(anchorCp.root);
+        if (this.anchorBlock) {
+          this.lastBackSyncedSlot = this.anchorBlock.message.slot;
+        } else {
+          this.anchorBlockRoot = anchorCp.root;
+          this.lastBackSyncedSlot = null;
+        }
+
+        this.backfillStartFromSlot = anchorCp.epoch * SLOTS_PER_EPOCH;
+        this.logger.info("BackfillSync - initializing from Checkpoint", {
+          root: toHexString(anchorCp.root),
+          epoch: anchorCp.epoch,
+        });
+      }
+
+      // Update from the backfilled sequence
+
+      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```

Comment:
Please remove all disablings of the rule `@typescript-eslint/strict-boolean-expressions` from this PR